### PR TITLE
feat(surveys): Make surveys site app native to posthog-js

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

Deprecate surveys site app, and make surveys part of posthog-js, as an optional loader

https://github.com/PostHog/feature-surveys

I opted not to make an interface for `PostHog` core, because as it turns out, there's some bundling magic happening which strips the type out of the js bundle, ensuring the size remains small.

Only when I do something with `PostHog` does it get bundled in.

I'd recommend checking all the code except what's in `extensions/surveys.ts` because that's just a copy from the site app.

A future PR might change this into a more extensible react based flow. Right now, the focus is making sure this can be loaded correctly.

(I need to write a few more loader tests, which I will soon, but wanted the PoC out for review)
...

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers

### CODE REVIEW

**Comment by neilkakkar** [comment]
This is part 1 of X.

This should be safe to go in as is, because until `/decide` starts returning surveys, no new ones will be triggered from `posthog-js`.

I'll test this manually, check we have it in the CDN, before making the change in app & figuring out cache busting rules.

**Comment by liyiy** [comment]
Surveys site app code will get updated relatively often, but now users will have to manually update their posthog js version each time to get any new updates? Hmmmm

**Comment by neilkakkar** [comment]
No, because the latest version is loaded from our servers. They can be on an older posthog-js version and still get the latest surveys, they're independent. Unless, they explicitly bundle the two together and pin it.

**Comment by liyiy** [comment]
> No, because the latest version is loaded from our servers. They can be on an older posthog-js version and still get the latest surveys, they're independent. Unless, they explicitly bundle the two together and pin it.

Great

**Review by liyiy — APPROVED** [review]

File: src/decide.ts [hunk 1]
```diff
@@ -74,6 +74,23 @@ export class Decide {
...
+
+        if (response['surveys'] && !surveysGenerator) {
+            loadScript(this.instance.get_config('api_host') + `/static/surveys.js`, (err) => {
+                if (err) {
+                    return console.error(`Could not load surveys script`, err)
+                }
+
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-ignore
+                window.generateSurveys(this.instance)
```
Comment (benjackwhite) [inline src/decide.ts:90]: Can we scope this a little more perhaps? `window.extendPostHogWithSurveys` or something? 

Should be incredibly low chance of a clash but still. "generateSurveys" is somewhat too generic

**Review by benjackwhite — CHANGES_REQUESTED** [review]
Overall looks good. I didn't dive into the actual "surveys" code much. I think as soon as possible we should consider moving this to some sort of lighweight frontend framework but thats for another time

File: src/extensions/surveys.ts [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+    callSurveys(posthog, true)
+
+    let currentUrl = location.href
+    if (location.href) {
+        setInterval(() => {
+            if (location.href !== currentUrl) {
+                currentUrl = location.href
+                callSurveys(posthog, false)
+            }
+        }, 1500)
```
Comment (benjackwhite) [inline src/extensions/surveys.ts:554]: Opinion incoming - this feels like something that should be in `posthog-js` - like `posthog.onActiveSurveysChanged()` or something. Or is it somehow unique to our GUI implementation?

Comment (neilkakkar) [reply src/extensions/surveys.ts:554]: Makes some sense, will skip for now though and have a think after the migration

File: src/decide.ts (hunk 1 above)
Comment (neilkakkar) [reply src/decide.ts:90]: Makes sense, will do 👍

**Review by benjackwhite — APPROVED** [review]

**Review by jurajmajerik — APPROVED** [review]

**Pull request merged**